lib/matplotlib/lines.py

```diff
@@ -36,24 +36,47 @@
 from matplotlib import _path
 
 
-def get_dash_pattern(style):
-    """
-    Given a dash pattern name from 'solid', 'dashed', 'dashdot' or
-    'dotted', returns the (offset, dashes) pattern.
+def _get_dash_pattern(style):
+    """Convert linestyle -> dash pattern
+
     """
-    if style == 'solid':
+    # go from short hand -> full strings
+    if is_string_like(style) and is_hashable(style):
+        style = ls_mapper.get(style, style)
+    # un-dashed styles
+    if style in ['solid', 'None']:
         offset, dashes = None, None
+    # dashed styles
     elif style in ['dashed', 'dashdot', 'dotted']:
         offset = 0
         dashes = tuple(rcParams['lines.{}_pattern'.format(style)])
+    #
     elif isinstance(style, tuple):
         offset, dashes = style
     else:
         raise ValueError('Unrecognized linestyle: %s' % str(style))
 
+    # normalize offset to be positive and shorter than the dash cycle
+    if dashes is not None and offset is not None:
+        offset %= sum(dashes)
```

Comment:
Don't know if one are supposed to be “bad user-proof”, but if `dashes` is (0, 0), won't it raise a `ZeroDivisionError`? (I know it's kind of a corner case though)